### PR TITLE
photocollage: init at 1.4.6

### PR DESCRIPTION
**Pull request by eliandoran**
## Description of changes

Graphical tool to make photo collage posters

Should also close https://github.com/NixOS/nixpkgs/pull/210567 .

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) (or backporting [23.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md) and [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

File: pkgs/by-name/ph/photocollage/package.nix [hunk 1]
```diff
@@ -0,0 +1,47 @@
+{ lib, python3, python3Packages, fetchFromGitHub,
```
Comment (AndersonTorres) [inline pkgs/by-name/ph/photocollage/package.nix:1]: 1. one attribute per line
2. remove python3

File: pkgs/by-name/ph/photocollage/package.nix [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+
+  meta = {
+    description = "Graphical tool to make photo collage posters";
+    homepage = "https://github.com/adrienverge/PhotoCollage";
+    license = lib.licenses.gpl2Plus;
+    maintainers = with lib.maintainers; [ eliandoran ];
+    platforms = [ "x86_64-linux" ];
+    mainProgram = "photocollage";
+  };
+
```
Comment (AndersonTorres) [inline pkgs/by-name/ph/photocollage/package.nix:46]:
```suggestion
```

File: pkgs/by-name/ph/photocollage/package.nix [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+    mkdir -p $out/share/icons/hicolor
+    cp -r ./data/icons/hicolor/* $out/share/icons/hicolor
+  '';
+
+  meta = {
+    description = "Graphical tool to make photo collage posters";
+    homepage = "https://github.com/adrienverge/PhotoCollage";
+    license = lib.licenses.gpl2Plus;
+    maintainers = with lib.maintainers; [ eliandoran ];
+    platforms = [ "x86_64-linux" ];
```
Comment (AndersonTorres) [inline pkgs/by-name/ph/photocollage/package.nix:43]:
```suggestion
```

There is nothing on this pkg that constrains it to Linux at AMD64

**Review by AndersonTorres — CHANGES_REQUESTED** [review]

File: pkgs/by-name/ph/photocollage/package.nix (hunk 3 above)
Comment (eliandoran) [reply pkgs/by-name/ph/photocollage/package.nix:43]: @AndersonTorres , you're right, although generally I put only the platforms I actually tested on.

**Comment by eliandoran** [comment]
@AndersonTorres , please re-review.

**Review by SuperSandro2000 — APPROVED** [review]

**Review by doronbehar — CHANGES_REQUESTED** [review]
Nice package :) Looks good overall.

File: pkgs/by-name/ph/photocollage/package.nix [hunk 4]
```diff
@@ -0,0 +1,51 @@
...
+, wrapGAppsHook }:
+
+python3Packages.buildPythonApplication {
+  pname = "photocollage";
+  version = "1.4.6";
+
+  src = fetchFromGitHub {
+    owner = "adrienverge";
+    repo = "PhotoCollage";
+    rev = "29453db0a2b8e30c1fdc6a062d405eb58d917029"; # upstream maintainer currently does not tag releases
```
[the remark this answers is not in the document]
Comment (eliandoran) [reply pkgs/by-name/ph/photocollage/package.nix]: @doronbehar , apparently they do... thanks for noticing! :)

File: pkgs/by-name/ph/photocollage/package.nix [hunk 5]
```diff
@@ -0,0 +1,52 @@
...
+    gettext
+    gobject-introspection
+    wrapGAppsHook
+  ];
+
+  postInstall = ''
+    install -D ./data/photocollage.desktop $out/share/applications/photocollage.desktop
+
+    mkdir -p $out/share/icons/hicolor
+    cp -r ./data/icons/hicolor/* $out/share/icons/hicolor
```
Comment (doronbehar) [inline pkgs/by-name/ph/photocollage/package.nix:41]: I'd be more accurate, readable and shorter to do this:

```suggestion
    # Based on the debian package's list of files. Link:
    # https://packages.debian.org/bookworm/all/photocollage/filelist
    install -Dm0644 ./data/photocollage.desktop $out/share/applications/photocollage.desktop
    install -Dm0644 ./data/photocollage.appdata.xml $out/share/appdata/photocollage.appdata.xml
    cp -r ./data/icons $out/share/icons
```

**Review by doronbehar — COMMENTED** [review]
Forgive me for the iterations, all of a sudden I noticed this detail and had to write something. You can also apply my suggestion via the web interface and I'll squash merge this if CI is green.

**Review by doronbehar — APPROVED** [review]

File: pkgs/by-name/ph/photocollage/package.nix [hunk 6]
```diff
@@ -0,0 +1,53 @@
...
+  ];
+
+  buildInputs = [
+    gdk-pixbuf
+  ];
+
+  nativeBuildInputs = [
+    gettext
+    gobject-introspection
+    wrapGAppsHook
```
Comment (doronbehar) [inline pkgs/by-name/ph/photocollage/package.nix:34]: One more issue I noticed when building locally, is that this input causes the binary to be wrapped twice. See solution in docs:

https://nixos.org/manual/nixpkgs/unstable/#ssec-gnome-common-issues-double-wrapped

**Review by doronbehar — CHANGES_REQUESTED** [review]
Almost perfect (again sorry for taking long to respond to changes).

File: pkgs/by-name/ph/photocollage/package.nix (hunk 6 above)
Comment (eliandoran) [reply pkgs/by-name/ph/photocollage/package.nix:34]: @doronbehar , should be OK now. How did you notice that the the binary was wrapped twice when building locally?

Comment (doronbehar) [reply pkgs/by-name/ph/photocollage/package.nix:34]: > @doronbehar , should be OK now. How did you notice that the the binary was wrapped twice when building locally?

I pressed <kbd>tab</kbd> for shell completion while I had in my shell line editor `./results/photocollage/bin/`.

**Review by doronbehar — APPROVED** [review]

**Pull request merged**